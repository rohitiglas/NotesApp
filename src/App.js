import React from 'react';
import {Text, View} from 'react-native';
export default function App() {
  let [movies, setMovies] = React.useState([]);

  React.useEffect(() => {
    fetch('/api/notes')
      .then(res => res.json())
      .then(data => setMovies(data.notes))
      .catch(error => console.log('Error fetching notes', error));
  }, []);

  return (
    <View style={{marginTop: 50, marginLeft: 10, marginRight: 10}}>
      {movies.map(movie => (
        <View style={{marginTop: 30}}>
          <Text style={{fontSize: 20}} key={movie.id}>
            {movie.title}
          </Text>
          <Text style={{fontSize: 10}}>
            {movie.body}
          </Text>
        </View>
      ))}
    </View>
  );
}
