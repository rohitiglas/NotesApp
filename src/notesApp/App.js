import React, {useEffect, useState} from 'react';
import {
  View,
  Text,
  TouchableOpacity,
  Alert,
  ActivityIndicator,
} from 'react-native';
import NotesList from './NotesList';
import AddEditNote from './AddEditNote';
import axios from 'axios';
const App = () => {
  const [notes, setNotes] = useState([]);
  const [note, setNote] = useState({});
  const [inputs, setInputs] = useState({title: '', body: ''});
  const [toggle, setToggle] = useState(false);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    getNotes();
  }, []);

  const getNotes = () => {
    setLoading(true);
    axios
      .get('/api/notes')
      .then(function (response) {
        setLoading(false);
        setNotes(response.data.notes);
      })
      .catch(function (error) {
        console.log(error);
      });
  };
  const getNote = id => {
    console.log("Note id",id);
    setLoading(true);

    axios
      .get(`/api/notes/${id}`)
      .then(function (response) {
        console.log("Note id Response",response);
        setNote(response.data.notes);
        setLoading(false);
        setToggle(true);
      })
      .catch(error => console.log('Note id Error"', error));



    // fetch(`/api/notes/${id}`)
    //   .then(res => res.json())
    //   .then(data => {
    //     setNote(data.notes);
    //     setLoading(false);
    //     setToggle(true);
    //   })
    //   .catch(error => console.log('Note not found', error));
  };
  const addNote = (title, body) => {
    if (!title || !body) {
      Alert.alert('', 'Please fill all the required input fields');
      return;
    }
    setLoading(true);
    axios.post('/api/notes', {
      title: title,
      body: body
    })
    .then(function (response) {
      console.log(response);
        setLoading(false);
        getNotes();
        setToggle(false);
        setInputs({title: '', body: ''});
        Alert.alert('', 'Note added successfully');
    })
    .catch(function (error) {
      console.log('Error adding note.', error);

        Alert.alert('', 'Error adding note.');
    });
    // fetch('/api/notes', {
    //   method: 'POST',
    //   body: JSON.stringify({
    //     title: title,
    //     body: body,
    //   }),
    // })
    //   .then(res => {
    //     console.log(res.json());
    //     setLoading(false);
    //     getNotes();
    //     setToggle(false);
    //     setInputs({title: '', body: ''});
    //     Alert.alert('', 'Note added successfully');
    //   })
    //   .catch(error => {
    //     console.log('Error adding note.', error);

    //     Alert.alert('', 'Error adding note.');
    //   });
  };

  const updateNote = (id, title, body) => {
    console.log("Update Note",title);
    console.log("Update Note Body",body);
    if (!title || !body) {
      Alert.alert('', 'Please fill all the required input fields');
      return;
    }
    setLoading(true);
    fetch(`/api/notes/${id}`, {
      method: 'PATCH',
      body: JSON.stringify({
        title: title,
        body: body,
      }),
    })
      .then(res => {
        getNotes();
        setToggle(false);
        setLoading(false);
        Alert.alert('', 'Note updated successfully.');
      })
      .catch(error => {
        console.log('Note not found', error);
        Alert.alert('', 'Error updating note.');
      });
  };

  const deleteNote = id => {
    console.log("SSLSLLSLSLSLSLLSLSLSLIDIDIDIIDID",id);
    setLoading(true);


    axios.delete(`/api/notes/${id}`)
    .then(function (response) {
      console.log("SSLSLLSLSLSLSLLSLSLSL",response);
      getNotes();
      setLoading(false);
      Alert.alert('', 'Note deleted successfully');
    })
    .catch(function (error) {
      console.log('Note not found', error);
      Alert.alert('', 'Error deleting note.');
    });



    // fetch(`/api/notes/${id}`, {method: 'DELETE'})
    //   .then(res => {
    //     getNotes();
    //     setLoading(false);
    //     Alert.alert('', 'Note deleted successfully');
    //   })
    //   .catch(error => {
    //     console.log('Note not found', error);
    //     Alert.alert('', 'Error deleting note.');
    //   });
  };

  const handleAdd = () => {
    setInputs({title: '', body: ''});
    setNote([]);
    setToggle(!toggle);
  };

  return (
    <View style={{flex: 1, marginTop: 50}}>
      <Text
        testID={'notes-title'}
        style={{
          color: '#1741db',
          textAlign: 'center',
          fontSize: 20,
          fontWeight: '700',
        }}>
        Notes
      </Text>
      {loading && (
        <View
          testID={'loader'}
          style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
          <ActivityIndicator size="large" />
        </View>
      )}
      <TouchableOpacity
        testID={'add-notes'}
        style={{
          borderWidth: 1,
          borderColor: 'rgba(0,0,0,0.2)',
          alignItems: 'center',
          justifyContent: 'center',
          width: 70,
          position: 'absolute',
          zIndex: 1,
          bottom: 10,
          right: 10,
          height: 70,
          backgroundColor: '#0b43e0',
          borderRadius: 100,
        }}
        onPress={handleAdd}>
        <Text
          style={{
            color: '#fff',
            alignSelf: 'center',
            fontSize: 30,
            fontWeight: '700',
          }}>
          {toggle ? 'X' : '+'}
        </Text>
      </TouchableOpacity>
      {!toggle && (
        <NotesList
          testId={'notes-list'}
          notes={notes}
          getNote={getNote}
          setToggle={setToggle}
          deleteNote={deleteNote}
        />
      )}
      {toggle && (
        <AddEditNote
          testId={'add-edit-page'}
          showModal={toggle}
          note={note}
          toggleModal={setToggle}
          addNote={addNote}
          updateNote={updateNote}
          inputs={inputs}
          setInputs={setInputs}
        />
      )}
    </View>
  );
};
export default App;
