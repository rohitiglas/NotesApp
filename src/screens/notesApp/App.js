import React, {useEffect, useState} from 'react';
import {View, Alert, Modal, TouchableOpacity, Text} from 'react-native';
// import SplashScreen from 'react-native-splash-screen';
import NotesList from './NotesList';
import AddEditNote from './addOrEdit/AddEditNote';
import axios from 'axios';
import SearchBar from './searchBar/SearchBar';
import Error from './errorScreen/Error';
import Loader from '../../components/Loader';
import FabButton from './FAB/FabButton';
import {_storeData, resetNavigation} from '../../asyncStorage/Local';
const App = ({navigation}) => {
  const [notes, setNotes] = useState([]);
  const [note, setNote] = useState({});
  const [inputs, setInputs] = useState({title: '', body: ''});
  const [toggle, setToggle] = useState(false);
  const [loading, setLoading] = useState(false);
  const [searchResult, setSearchResult] = useState({});
  const [query, setQuery] = useState('');

  React.useLayoutEffect(() => {
    navigation.setOptions({
      title: 'Notes', //Set Header Title
      headerStyle: {
        backgroundColor: '#66CCCC', //Set Header color
      },
      headerTintColor: '#fff', //Set Header text color
      headerTitleStyle: {
        fontWeight: 'bold', //Set Header text style
        fontSize: 20,
      },
      headerLeft: () => (
        <TouchableOpacity onPress={() => Logout()} style={{marginLeft: 10}}>
          <Text style={{color: 'white', fontSize: 16, fontWeight: 'bold'}}>
            Logout
          </Text>
        </TouchableOpacity>
      ),
      headerRight: () => {},
    });
  }, [navigation]);

  const Logout = () => {
    resetNavigation(navigation, 'Login');
    _storeData('');
  };

  useEffect(() => {
    getNotes();
  }, []);

  const handleSearch = text => {
    const formattedQuery = text.toLowerCase();
    const data = searchResult.filter(user => {
      return contains(user, formattedQuery);
    });
    setQuery(text);
    setNotes(data);
  };
  const clearSearch = () => {
    setQuery('');
    setNotes(searchResult);
    // getNotes();
  };
  const contains = ({title, body}, query) => {
    return !!(
      title.toLowerCase().includes(query) || body.toLowerCase().includes(query)
    );
  };

  const getNotes = () => {
    setLoading(true);
    axios
      .get('/api/notes')
      .then(function (response) {
        setLoading(false);
        setNotes(response.data.notes);
        setSearchResult(response.data.notes);
      })
      .catch(function (error) {
        setLoading(false);
        console.log(error);
      });
  };
  const getNote = id => {
    setLoading(true);

    axios
      .get(`/api/notes/${id}`)
      .then(function (response) {
        setNote(response.data.notes);
        setLoading(false);
        setToggle(true);
      })
      .catch(error => {
        console.log('Note id Error"', error);
        setLoading(false);
      });
  };
  const addNote = (title, body) => {
    console.log('Add Notes Start', title);
    if (!title || !body) {
      Alert.alert('', 'Please fill all the required input fields');
      return;
    }
    setLoading(true);
    axios
      .post('/api/notes', {
        title: title,
        body: body,
      })
      .then(function (response) {
        console.log('Add Notes Response', response);
        setLoading(false);
        getNotes();
        setToggle(false);
        setInputs({title: '', body: ''});
        Alert.alert('', 'Note added successfully');
      })
      .catch(function (error) {
        console.log('Add Notes Error', error);
        Alert.alert('', 'Error adding note.');
        setLoading(false);
      });
  };

  const updateNote = (id, title, body) => {
    if (!title || !body) {
      Alert.alert('', 'Please fill all the required input fields');
      return;
    }
    setLoading(true);
    axios
      .patch(`/api/notes/${id}`, {
        title: title,
        body: body,
      })
      .then(function (response) {
        getNotes();
        setToggle(false);
        setLoading(false);
        Alert.alert('', 'Note updated successfully.');
      })
      .catch(function () {
        Alert.alert('', 'Error updating note.');
        setLoading(false);
      });
  };

  const deleteNote = id => {
    Alert.alert(
      'Notes',
      'Do you want to delete the note',
      [
        {
          text: 'NO',
          onPress: () => {},
        },
        {
          text: 'YES',
          onPress: () => {
            setLoading(true);

            axios
              .delete(`/api/notes/${id}`)
              .then(function (response) {
                getNotes();
                setLoading(false);
                Alert.alert('', 'Note deleted successfully');
              })
              .catch(function (error) {
                Alert.alert('', 'Error deleting note.');
                setLoading(false);
              });
          },
        },
      ],
      {cancelable: false},
    );
  };

  const handleAdd = () => {
    setInputs({title: '', body: ''});
    setNote([]);
    setToggle(!toggle);
  };

  return (
    <View style={{flex: 1, marginTop: 10}}>
      <SearchBar
        query={query}
        onChangeText={handleSearch}
        clearText={clearSearch}
      />
      <Loader isLoading={loading} />
      {notes && notes.length === 0 ? (
        <Error />
      ) : (
        !toggle && (
          <NotesList
            testId={'notes-list'}
            notes={notes}
            getNote={getNote}
            setToggle={setToggle}
            deleteNote={deleteNote}
          />
        )
      )}
      {!toggle && <FabButton handleAdd={handleAdd} toggle={toggle} />}
      <Modal animationType="slide" transparent={true} visible={toggle}>
        <AddEditNote
          testId={'add-edit-page'}
          showModal={toggle}
          note={note}
          closeModal={() => setToggle(false)}
          toggleModal={setToggle}
          addNote={addNote}
          updateNote={updateNote}
          inputs={inputs}
          setInputs={setInputs}
        />
      </Modal>
    </View>
  );
};
export default App;
