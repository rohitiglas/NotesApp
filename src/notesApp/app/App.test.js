import {Alert} from 'react-native';
import React from 'react';
import {render, cleanup, fireEvent, flushMicrotasksQueue, waitForElement} from '@testing-library/react-native';
import App from '../App';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

// jest.mock('../AddEditNote');
jest.spyOn(Alert, 'alert');

let mock = new MockAdapter(axios);

describe('App screen', () => {
  let wrapper;
  beforeEach(() => {
    mock.onGet('/api/notes').reply(200, {
      notes: [{title: 'Rohit', body: 'Bansal', id: '10'}],
    });
    mock.onDelete(`/api/notes/10`).reply(200, {});
    mock.onGet(`/api/notes/10`).reply(200, {notes:{title: 'Rohit', body: 'Bansal', id: '10'}});
    mock.onPatch(`/api/notes/10`,{title:'Rohit',description:'Body'}).reply(200, {});
    wrapper = render(<App />);
  });
  afterEach(() => {
    cleanup();
    wrapper = null;
  });
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should render notes list',async ()=>{
    const {getByTestId, queryByTestId,debug} = wrapper;
    expect(getByTestId('notes-title')).toBeDefined();
    expect(getByTestId('add-notes')).toBeDefined();
    await flushMicrotasksQueue();
    expect(getByTestId('list-view')).toBeDefined();
    expect(queryByTestId('no-notes')).toBeNull();
    expect(getByTestId('list-data')).toBeDefined();
    await flushMicrotasksQueue();
    // debug();
    expect(getByTestId('title').children[0]).toBe('Rohit');
  });
  it('should delete note when delete button press',async ()=>{
    const {getByTestId, queryByTestId,debug} = wrapper;
    expect(getByTestId('notes-title')).toBeDefined();
    expect(getByTestId('add-notes')).toBeDefined();
    await flushMicrotasksQueue();
    expect(getByTestId('list-view')).toBeDefined();
    expect(queryByTestId('no-notes')).toBeNull();
    expect(getByTestId('list-data')).toBeDefined();
    const deleteNote=getByTestId('delete-button');
    expect(deleteNote).toBeDefined();
    fireEvent.press(deleteNote);
    await flushMicrotasksQueue();
    // debug();
    expect(mock.history.delete.length).toBe(1);
  });
  it('should update api called when update button press',async ()=>{
    const {getByTestId, queryByTestId,debug} = wrapper;
    expect(getByTestId('notes-title')).toBeDefined();
    expect(getByTestId('add-notes')).toBeDefined();
    await flushMicrotasksQueue();
    expect(getByTestId('list-view')).toBeDefined();
    expect(queryByTestId('no-notes')).toBeNull();
    expect(getByTestId('list-data')).toBeDefined();
    const updateNote=getByTestId('update-button');
    expect(updateNote).toBeDefined();
    fireEvent.press(updateNote);
    await flushMicrotasksQueue();
    console.log("Url is",mock.history.get);
    expect(mock.history.get.length).toBe(6);
    await flushMicrotasksQueue();
    debug();
    const title=getByTestId('input-title');
    const description=getByTestId('input-description');
    expect(title.props.value).toBe('Rohit');
    expect(description.props.value).toBe('Bansal');
  });
  it('should not render notes list and Text no notes available',async ()=>{
    mock.onGet('/api/notes').reply(500, {
    });
    const {getByTestId, queryByTestId,debug} = wrapper;
    expect(getByTestId('notes-title')).toBeDefined();
    expect(getByTestId('add-notes')).toBeDefined();
    await flushMicrotasksQueue();
    expect(getByTestId('list-view')).toBeDefined();
    expect(queryByTestId('list-data')).toBeNull();
    expect(getByTestId('no-notes')).toBeDefined();
  });
  it('should show Alert message on Edit/Add notes page when title and description empty on Add Note button click', async function () {
    const {getByTestId, queryByTestId,debug} = wrapper;
    expect(getByTestId('notes-title')).toBeDefined();
    expect(getByTestId('add-notes')).toBeDefined();
    // await fireEvent(getByTestId('add-notes'), 'onClick');
    await fireEvent.press(getByTestId('add-notes'));
    await flushMicrotasksQueue();
    expect(getByTestId('add-edit-page')).toBeDefined();
    const title=getByTestId('input-title');
    const description=getByTestId('input-description');
    const submitButton=getByTestId('submit-button');
    fireEvent.changeText(title, '');
    fireEvent.changeText(description, '');
    await fireEvent.press(submitButton);
    expect(Alert.alert).toHaveBeenCalledWith('','Please fill all the required input fields')
    expect(wrapper).toMatchSnapshot();
  });
  it('should show Edit/Add notes page when button click', async function () {
    const {getByTestId, queryByTestId,debug} = wrapper;
    expect(getByTestId('notes-title')).toBeDefined();
    expect(getByTestId('add-notes')).toBeDefined();
    // await fireEvent(getByTestId('add-notes'), 'onClick');
    await fireEvent.press(getByTestId('add-notes'));
    await flushMicrotasksQueue();
    expect(getByTestId('add-edit-page')).toBeDefined();
    const title=getByTestId('input-title');
    const description=getByTestId('input-description');
    const submitButton=getByTestId('submit-button');
    fireEvent.changeText(title, 'My Notes');
    fireEvent.changeText(description, 'This is my First Note');
    await fireEvent.press(submitButton);
    // mock.onPost('/api/notes',{ title: 'My Notes',description: 'This is my First Note'}).reply(200, {
      
    // });
    expect(mock.history.post.length).toBe(1);
    expect(mock.history.post[0].data).toBe(JSON.stringify({ title: 'My Notes', body: "This is my First Note" }));
    expect(wrapper).toMatchSnapshot();
  });
});