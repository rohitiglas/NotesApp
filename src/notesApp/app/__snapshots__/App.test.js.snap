// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App screen renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginTop": 50,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#1741db",
        "fontSize": 20,
        "fontWeight": "700",
        "textAlign": "center",
      }
    }
    testID="notes-title"
  >
    Notes
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="loader"
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0b43e0",
        "borderColor": "rgba(0,0,0,0.2)",
        "borderRadius": 100,
        "borderWidth": 1,
        "bottom": 10,
        "height": 70,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 10,
        "width": 70,
        "zIndex": 1,
      }
    }
    testID="add-notes"
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 30,
          "fontWeight": "700",
        }
      }
    >
      +
    </Text>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "flex": 1,
        "marginTop": 0,
      }
    }
    testID="list-view"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#d90617",
            "fontSize": 30,
            "fontWeight": "700",
            "textAlign": "center",
          }
        }
        testID="no-notes"
      >
        No Notes available
      </Text>
    </View>
  </RCTSafeAreaView>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "backgroundColor": "#2c3e50",
          "flex": 1,
        }
      }
      testID="add-edit-page"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexGrow": 1,
            "justifyContent": "center",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          maxLength={100}
          multiline={true}
          numberOfLines={2}
          onChangeText={[Function]}
          placeholder="Title"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 20,
              "color": "#2c3e50",
              "fontSize": 20,
              "height": 80,
              "marginBottom": 20,
              "padding": 10,
              "paddingTop": 20,
              "width": "80%",
            }
          }
          testID="input-title"
          underlineColorAndroid="transparent"
        />
        <TextInput
          allowFontScaling={true}
          maxLength={1000}
          multiline={true}
          numberOfLines={10}
          onChangeText={[Function]}
          placeholder="Body"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 20,
              "color": "#2c3e50",
              "fontSize": 20,
              "height": 200,
              "marginBottom": 20,
              "padding": 10,
              "paddingTop": 20,
              "textAlignVertical": "top",
              "width": "80%",
            }
          }
          testID="input-description"
          underlineColorAndroid="transparent"
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2980b6",
              "borderRadius": 20,
              "opacity": 1,
              "paddingVertical": 15,
              "width": "80%",
            }
          }
          testID="submit-button"
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontWeight": "700",
                "textAlign": "center",
              }
            }
          >
            Add Note
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#000",
              "borderRadius": 25,
              "height": 50,
              "justifyContent": "center",
              "marginTop": 20,
              "opacity": 1,
              "position": "absolute",
              "right": 20,
              "top": 40,
              "width": 50,
            }
          }
          testID="close-button"
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontWeight": "700",
                "textAlign": "center",
              }
            }
          >
            X
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`App screen should show Alert message on Edit/Add notes page when title and description empty on Add Note button click 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginTop": 50,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#1741db",
        "fontSize": 20,
        "fontWeight": "700",
        "textAlign": "center",
      }
    }
    testID="notes-title"
  >
    Notes
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0b43e0",
        "borderColor": "rgba(0,0,0,0.2)",
        "borderRadius": 100,
        "borderWidth": 1,
        "bottom": 10,
        "height": 70,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 10,
        "width": 70,
        "zIndex": 1,
      }
    }
    testID="add-notes"
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 30,
          "fontWeight": "700",
        }
      }
    >
      X
    </Text>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "backgroundColor": "#2c3e50",
          "flex": 1,
        }
      }
      testID="add-edit-page"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexGrow": 1,
            "justifyContent": "center",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          maxLength={100}
          multiline={true}
          numberOfLines={2}
          onChangeText={[Function]}
          placeholder="Title"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 20,
              "color": "#2c3e50",
              "fontSize": 20,
              "height": 80,
              "marginBottom": 20,
              "padding": 10,
              "paddingTop": 20,
              "width": "80%",
            }
          }
          testID="input-title"
          underlineColorAndroid="transparent"
          value=""
        />
        <TextInput
          allowFontScaling={true}
          maxLength={1000}
          multiline={true}
          numberOfLines={10}
          onChangeText={[Function]}
          placeholder="Body"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 20,
              "color": "#2c3e50",
              "fontSize": 20,
              "height": 200,
              "marginBottom": 20,
              "padding": 10,
              "paddingTop": 20,
              "textAlignVertical": "top",
              "width": "80%",
            }
          }
          testID="input-description"
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2980b6",
              "borderRadius": 20,
              "opacity": 1,
              "paddingVertical": 15,
              "width": "80%",
            }
          }
          testID="submit-button"
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontWeight": "700",
                "textAlign": "center",
              }
            }
          >
            Add Note
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#000",
              "borderRadius": 25,
              "height": 50,
              "justifyContent": "center",
              "marginTop": 20,
              "opacity": 1,
              "position": "absolute",
              "right": 20,
              "top": 40,
              "width": 50,
            }
          }
          testID="close-button"
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontWeight": "700",
                "textAlign": "center",
              }
            }
          >
            X
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`App screen should show Edit/Add notes page when button click 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginTop": 50,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#1741db",
        "fontSize": 20,
        "fontWeight": "700",
        "textAlign": "center",
      }
    }
    testID="notes-title"
  >
    Notes
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="loader"
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0b43e0",
        "borderColor": "rgba(0,0,0,0.2)",
        "borderRadius": 100,
        "borderWidth": 1,
        "bottom": 10,
        "height": 70,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 10,
        "width": 70,
        "zIndex": 1,
      }
    }
    testID="add-notes"
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 30,
          "fontWeight": "700",
        }
      }
    >
      X
    </Text>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "backgroundColor": "#2c3e50",
          "flex": 1,
        }
      }
      testID="add-edit-page"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexGrow": 1,
            "justifyContent": "center",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          maxLength={100}
          multiline={true}
          numberOfLines={2}
          onChangeText={[Function]}
          placeholder="Title"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 20,
              "color": "#2c3e50",
              "fontSize": 20,
              "height": 80,
              "marginBottom": 20,
              "padding": 10,
              "paddingTop": 20,
              "width": "80%",
            }
          }
          testID="input-title"
          underlineColorAndroid="transparent"
          value="My Notes"
        />
        <TextInput
          allowFontScaling={true}
          maxLength={1000}
          multiline={true}
          numberOfLines={10}
          onChangeText={[Function]}
          placeholder="Body"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 20,
              "color": "#2c3e50",
              "fontSize": 20,
              "height": 200,
              "marginBottom": 20,
              "padding": 10,
              "paddingTop": 20,
              "textAlignVertical": "top",
              "width": "80%",
            }
          }
          testID="input-description"
          underlineColorAndroid="transparent"
          value="This is my First Note"
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2980b6",
              "borderRadius": 20,
              "opacity": 1,
              "paddingVertical": 15,
              "width": "80%",
            }
          }
          testID="submit-button"
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontWeight": "700",
                "textAlign": "center",
              }
            }
          >
            Add Note
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#000",
              "borderRadius": 25,
              "height": 50,
              "justifyContent": "center",
              "marginTop": 20,
              "opacity": 1,
              "position": "absolute",
              "right": 20,
              "top": 40,
              "width": 50,
            }
          }
          testID="close-button"
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontWeight": "700",
                "textAlign": "center",
              }
            }
          >
            X
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
